kqueue_test.go

```diff
@@ -0,0 +1,71 @@
+package fsnotify
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"sync"
+	"testing"
+)
+
+func consumeEvent(t *testing.T, watcher *Watcher, wg *sync.WaitGroup) {
+	defer func() {
+		//time.Sleep(time.Second)
+		watcher.Close()
+		wg.Done()
+	}()
+	log.Println("Came here")
+	notifications := 0
+	maxNotifications := 1
+	for {
+		if notifications >= maxNotifications {
+			log.Println("Reached max notifications")
+			return
+		}
+		select {
+		case event := <-watcher.Events:
+			log.Println("event:", event)
+			notifications++
+			continue
+		case err := <-watcher.Errors:
+			log.Println("error:", err)
+		}
+	}
+}
+
+func writeFiles(dir string) {
+	index := 0
+	for {
+		ioutil.TempFile(dir, "example")
+		ioutil.TempFile(dir, "example")
+		if index > 10 {
+			return
+		}
+		index++
+	}
+}
+func TestWatcherClose(t *testing.T) {
```

Comment:
this test could be made much simpler. We just need the watcher to be listening on some file, modify the file and then close the watcher without listening on either the errors or events channels.